bundles/org.eclipse.text/src/org/eclipse/jface/text/AbstractDocument.java

```diff
@@ -97,20 +97,20 @@ static private class RegisteredReplace {
 	/** The document's line tracker */
 	private ILineTracker fTracker;
 	/** The registered document listeners */
-	private ListenerList<IDocumentListener> fDocumentListeners;
+	private final ListenerList<IDocumentListener> fDocumentListeners= new ListenerList<>(ListenerList.IDENTITY);
 	/** The registered pre-notified document listeners */
-	private ListenerList<IDocumentListener> fPrenotifiedDocumentListeners;
+	private final ListenerList<IDocumentListener> fPrenotifiedDocumentListeners= new ListenerList<>(ListenerList.IDENTITY);
 	/** The registered document partitioning listeners */
-	private ListenerList<IDocumentPartitioningListener> fDocumentPartitioningListeners;
+	private final ListenerList<IDocumentPartitioningListener> fDocumentPartitioningListeners= new ListenerList<>(ListenerList.IDENTITY);
```

Comment:
This field wasn't quite effectively final as there are checks for `== null` in the class. I can't see how they could be hit and it is the only one of the `ListenerList`s to have the extra null check. But it may have implied that at some point there were listeners installed and changes fired before. Or it may have been defensive programming.